File: plugins/ads/source/iOS/Plugins/Ads/AdsManager.m
//
//  AdsManager.m
//  Ads
//
//  Created by Arturs Sosins on 3/15/14.
//  Copyright (c) 2014 Gideros Mobile. All rights reserved.
//

#import "AdsManager.h"

@implementation AdsManager
-(id)init{
    self.adViews_ = [NSMutableDictionary dictionary];
    return self;
}
-(NSObject*)get:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        return [state getObject];
    return nil;
}
-(AdsState*)getState:(NSString*)type{
    return [self.adViews_ objectForKey:type];
}
-(void)set:(NSObject*)object forType:(NSString*)type{
    AdsState *state = [[AdsState alloc] init:object withType:type];
    [self.adViews_ setObject:state forKey:type];
}
-(void)set:(NSObject*)object forType:(NSString*)type withListener:(AdsStateChangeListener*)listener{
    AdsState *state = [[AdsState alloc] init:object withType:type];
    [state setListener:listener];
    [self.adViews_ setObject:state forKey:type];
}
-(void)setObject:(NSObject*)object forType:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state setObject:object];
}
-(void)setListener:(AdsStateChangeListener*)listener forType:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state setListener:listener];
}
-(void)setAutoKill:(BOOL)autokill forType:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state setAutokill:autokill];
}
-(void)remove:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state destruct];
    [self.adViews_ removeObjectForKey:type];
}
-(void)hide:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state hide];
}
-(void)destroy{
    for(NSString *key in self.adViews_)
    {
        AdsState *state = [self.adViews_ objectForKey:key];
        [state destroy];
        [state destruct];
        [state release];
    }
}

-(void)reset:(NSString*)type{
    [self reset:type andKill:true];
}
-(void)reset:(NSString*)type andKill:(BOOL)kill{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state reset];
    if(kill)
        [self remove:type];
}

-(void)show:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state show];
}
-(void)load:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        [state load];
}
-(BOOL)isReady:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        return[state isReady];
    return false;
}
-(BOOL)isLoaded:(NSString*)type{
    AdsState *state = [self.adViews_ objectForKey:type];
    if(state != nil)
        return[state isLoaded];
    return false;
}
@end

File: plugins/ads/source/Android/src/com/giderosmobile/android/plugins/ads/AdsManager.java
package com.giderosmobile.android.plugins.ads;

import java.util.Enumeration;
import java.util.Hashtable;

public class AdsManager {
	
	private Hashtable<String, AdsState> adViews;
	
	public AdsManager(){
		adViews = new Hashtable<String, AdsState>();
	}
	
	public Object get(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			return adViews.get(type).getObject();
		return null;
	}
	
	public AdsState getState(String type){
		if(adViews != null && type != null)
			return adViews.get(type);
		return null;
	}
	
	public void set(Object adObject, String type){
		if(adViews != null && type != null)
		{
			AdsState adState = new AdsState(adObject, type);
			adViews.put(type, adState);
		}
	}
	
	public void set(Object adObject, String type, AdsStateChangeListener l){
		if(adViews != null && type != null)
		{
			AdsState adState = new AdsState(adObject, type);
			adState.setListener(l);
			adViews.put(type, adState);
		}
	}
	
	public void setObject(String type, Object o){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).setObject(o);
	}
	
	public void setListener(String type, AdsStateChangeListener l){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).setListener(l);
	}
	
	public void setAutoKill(String type, boolean kill){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).setAutoKill(kill);
	}

	public void setPreLoad(String type, boolean preload){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).setPreLoad(preload);
	}
	
	public void remove(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.remove(type);
	}
	
	public void hide(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).hide();
	}
	
	public void destroy(){
		if(adViews != null)
		{
			Enumeration<String> enumKey = adViews.keys();
			while(enumKey.hasMoreElements()) {
				String key = enumKey.nextElement();
				adViews.get(key).destroy();
				adViews.remove(key);
			}
			adViews = null;
		}
	}
	
	public void reset(String type){
		reset(type, true);
	}
	
	public void reset(String type, boolean delete){
		if(adViews != null && type != null && adViews.get(type) != null)
		{
			adViews.get(type).reset(delete);
			if(delete)
				adViews.remove(type);
		}
	}
	
	public void show(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).show();
	}
	
	public void load(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			adViews.get(type).load();
	}
	
	public boolean isReady(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			return adViews.get(type).isReady();
		return false;
	}
	
	public boolean isLoaded(String type){
		if(adViews != null && type != null && adViews.get(type) != null)
			return adViews.get(type).isLoaded();
		return false;
	}
}
